src/mqtt/pairing.cpp

```diff
@@ -0,0 +1,75 @@
+// (C) Copyright 2025, SECO Mind Srl
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "astarte_device_sdk/mqtt/pairing.hpp"
+
+#include <cpr/cpr.h>
+#include <spdlog/spdlog.h>
+
+#include <format>
+#include <nlohmann/json.hpp>
+#include <string>
+#include <string_view>
+
+#include "ada.h"
+
+using json = nlohmann::json;
+
+namespace AstarteDeviceSdk {
+
+auto PairingApi::create_pairing_url(std::string_view astarte_base_url) -> ada::url_aggregator {
```

Comment:
I would keep the private method at the bottom of the file, for consistency with the rest of the code-base